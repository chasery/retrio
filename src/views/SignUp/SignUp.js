import React from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import Header from '../../components/Header/Header';
import SignUpForm from '../../components/SignUpForm/SignUpForm';
import './SignUp.css';

function SignUp(props) {
  const location = useLocation();
  const history = useHistory();

  const handleSignUpSuccess = () => {
    const destination = (location.state || {}).from || '/racks';

    history.push(destination);
  };

  return (
    <>
      <Header />
      <main role='main'>
        <section className='SignUp'>
          <div className='SignUp__wrapper'>
            <SignUpForm onSignUpSuccess={handleSignUpSuccess} />
          </div>
        </section>
      </main>
    </>
  );
}

export default SignUp;
