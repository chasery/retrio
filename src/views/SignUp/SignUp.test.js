import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignUp from './SignUp';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: '/sign-up',
  }),
}));

describe('SignUp component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<SignUp />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
