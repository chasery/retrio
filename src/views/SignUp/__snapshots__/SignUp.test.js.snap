// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component renders without error 1`] = `
<Fragment>
  <Header />
  <main
    role="main"
  >
    <section
      className="SignUp"
    >
      <div
        className="SignUp__wrapper"
      >
        <SignUpForm
          onSignUpSuccess={[Function]}
        />
      </div>
    </section>
  </main>
</Fragment>
`;
